### PR TITLE
Feature/lista studenti

### PR DESCRIPTION
**Pull request by Razvan-Joita**

### CODE REVIEW

File: src/main/UI/modificareStudent.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        </div>
+        <button class="addStudentButton" onclick="addStudentForm()">Adaugare student</button>
+    </div>
+
+</div>
+<script src="script.js"></script>
+</body>
+</html>
+
+<!-----------Delete--------------------------->
```
Comment (Piciorus) [inline src/main/UI/modificareStudent.html]: you can remove comment code

File: src/main/UI/script.js [hunk 2]
```diff
@@ -0,0 +1,264 @@
+// //Sterge : mock data
+// const studentDataList = [
```
Comment (Piciorus) [inline src/main/UI/script.js]: remove if it is not used

**Comment by DragosPetri** [comment]
Tests fail in CI pipeline

File: src/main/UI/style.css [hunk 3]
```diff
@@ -0,0 +1,151 @@
+/*
```
Comment (Piciorus) [inline src/main/UI/style.css]: comment code?

File: src/main/java/org/example/Main.java [hunk 4]
```diff
@@ -3,6 +3,7 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
+
```
Comment (Piciorus) [inline src/main/java/org/example/Main.java]: remove empty line

File: src/main/java/org/example/controllers/StudentController.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+        Students addedStudent = studentService.addStudent(
+                firstName, lastName, cnp, birthDate, studyYear, studyLevel, fundingForm, graduatedHighSchool);
+
+        return ResponseEntity.status(HttpStatus.CREATED).body(addedStudent);
+    }
+
+
+
+//    @PostMapping
```
Comment (Piciorus) [inline src/main/java/org/example/controllers/StudentController.java]: do we need this code?

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1,7 +1,18 @@
-spring.jpa.hibernate.ddl-auto=update
-spring.datasource.url = jdbc:mysql://localhost:3300/proiectcol?allowPublicKeyRetrieval=true&useSSL=false
-spring.datasource.username = colectiv
-spring.datasource.password = colectiv
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/Students
+spring.datasource.username=root
```
Comment (Piciorus) [inline src/main/resources/application.properties:3]: why you changed the credentials for datasource?

**Review by Piciorus — CHANGES_REQUESTED** [review]

File: src/main/UI/style.css [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+.selectstyle {
+    padding: 8px;
+    margin-bottom: 16px;
+    border: 1px solid #fffdfd;
+    border-radius: 5px;
+    background: white;
+}
+
+/*#student-list {*/
```
Comment (Kl3igrec) [inline src/main/UI/style.css]: Remove it if you don't use it 🍡

File: src/main/java/org/example/controllers/StudentController.java [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+//            @RequestParam String fundingForm,
+//            @RequestParam String graduatedHighSchool) {
+//
+//        Students addedStudent = studentService.addStudent(
+//                firstName, lastName, cnp, birthDate, studyYear, studyLevel, fundingForm, graduatedHighSchool);
+//
+//        return ResponseEntity.status(HttpStatus.CREATED).body(addedStudent);
+//    }
+
+    @PostMapping(value = "/addStudent")
```
Comment (Kl3igrec) [inline src/main/java/org/example/controllers/StudentController.java]: This isn't restful, remove /addStudent

File: src/main/java/org/example/controllers/StudentController.java [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+        try {
+            studentService.deleteStudent(id);
+            return ResponseEntity.noContent().build();
+        } catch (EntityNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+    }
+
```
Comment (Kl3igrec) [inline src/main/java/org/example/controllers/StudentController.java:75]: Empty lines ?

File: src/main/java/org/example/services/StudentService.java [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+    @Autowired
+    private StudentRepository studentRepository;
+
+    private static final int CNP_LENGTH = 13;
+
+    private boolean isValidCnpLength(String cnp) {
+        return cnp != null && cnp.length() == CNP_LENGTH;
+    }
+
+    public List<Students> getAllStudents() {
```
Comment (Kl3igrec) [inline src/main/java/org/example/services/StudentService.java]: Please make a Paginated Get <3

File: src/main/java/org/example/services/StudentService.java [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+        return studentRepository.findById(id);
+    }
+
+    public Optional<Students> updateStudent(int studentId, Students updatedStudent) {
+        Optional<Students> existingStudent = studentRepository.findById(studentId);
+
+        if (existingStudent.isPresent()) {
+            Students savedStudent = existingStudent.get();
+
+            // Use setters directly for brevity
```
Comment (Kl3igrec) [inline src/main/java/org/example/services/StudentService.java]: ?????

File: src/main/java/org/example/services/StudentService.java [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+        student.setFirstName(firstName);
+        student.setLastName(lastName);
+        student.setCnp(cnp);
+        student.setBirthDate(birthDate);
+        student.setStudyYear(studyYear);
+        student.setStudyLevel(studyLevel);
+        student.setFundingForm(fundingForm);
+        student.setGraduatedHighSchool(graduatedHighSchool);
+
+        if (isValidCnpLength(student.getCnp())) {
```
Comment (Kl3igrec) [inline src/main/java/org/example/services/StudentService.java:87]: SOLID ? Please move this into a different method

File: src/main/java/org/example/controllers/StudentController.java [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+//        if (isValidCnpLength(student.getCnp())) {
+//            Students addedStudent = studentService.addStudent(student);
+//            return ResponseEntity.status(HttpStatus.CREATED).body(addedStudent);
+//        } else {
+//            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Invalid CNP length. CNP must be exactly 13 digits.");
+//        }
+//    }
+
+
+    private boolean isValidCnpLength(String cnp) {
```
Comment (Kl3igrec) [inline src/main/java/org/example/controllers/StudentController.java]: Are you using this?

**Review by DragosPetri — CHANGES_REQUESTED** [review]
Fix tests.

**Pull request closed without merging**